src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
You're right, you can't configure intervals with `xmpp_conn_set_flags()`. But I don't see tcp keepalives as the feature of xmpp library that requires new dedicated interface. Moreover, changing new interface requires changing of library's major version (this is not problem at the moment as major version should be changed in new release).

As I quickly googled, you can't change keepalive intervals on windows and interface should be compatible with this system.

There is a solution that would be useful not only in your case: some interface that gives access to the socket's file descriptor. So you can set any options you want in your program and use this descriptor in select(2) along with your own descriptors. The problem is that `int xmpp_conn_get_sock()` is not scalable solution. 2nd problem is in windows support again: correct code should use `SOCKET` instead of `int`.

As you see, it's not easy task to find an elegant solution.